pkg/cmd/cli/cmd/build.go

```diff
@@ -181,6 +181,6 @@ $ s2i build . centos/ruby-22-centos7 hello-world-app
 	buildCmd.Flags().BoolVarP(&(cfg.ForceCopy), "copy", "c", false, "Use local file system copy instead of git cloning the source url")
 	buildCmd.Flags().StringVar(&(cfg.RuntimeImage), "runtime-image", "", "Image that will be used as the base for the runtime image")
 	buildCmd.Flags().VarP(&(cfg.RuntimeArtifacts), "runtime-artifact", "a", "Specify a file or directory to be copied from the builder to the runtime image")
-
+	buildCmd.Flags().StringVarP(&(cfg.Network), "network", "", "", "Specify the default Docker Network name to be used in build process")
```

Comment:
unless i'm missing something, i think you can just set cfg.DockerNetworkMode w/ this flag.  The field name isn't ideal, but then you don't need the rest of the code changes, right?